discovery-provider/plugins/notifications/src/web.ts

```diff
@@ -0,0 +1,78 @@
+import webpush from 'web-push'
+import { logger } from './logger'
+import { Knex } from 'knex'
+import { Browser, UserNotificationSettings, WebPush } from './processNotifications/mappers/userNotificationSettings'
+
+// setup and configuration
+let webPushIsConfigured = false
+const vapidKeys = {
+    publicKey: process.env.BROWSER_PUSH_VAPID_PUBLIC_KEY,
+    privateKey: process.env.BROWSER_PUSH_VAPID_PRIVATE_KEY
+  }
+  const browserPushGCMAPIKey = process.env.BROWSER_PUSH_GCM_API_KEY
+  if (vapidKeys.publicKey && vapidKeys.privateKey && browserPushGCMAPIKey) {
+    webpush.setGCMAPIKey(browserPushGCMAPIKey)
+    webpush.setVapidDetails(
+      'mailto:contact@audius.co', // taken from identity
+      vapidKeys.publicKey,
+      vapidKeys.privateKey
+    )
+    // set config to true
+    webPushIsConfigured = true
+  } else {
+    logger.warn(
+      'Web Push is not configured. Browser Push API Notifs will NOT be sent'
+    )
+  }
+
+// utility functions
+export const sendBrowserNotification = async (settings: UserNotificationSettings, userId: number, title: string, message: string) => {
+    let numSentNotifs = 0
+  
+    try {
+      if (!webPushIsConfigured) return numSentNotifs
+      const userBrowserSettings = await settings.getUserBrowserSettings([userId])
+      const userNotificationSettings = userBrowserSettings[userId]
+
+      const isWebPush = (browser: Browser): boolean => {
+        return (browser as WebPush).p256dhKey !== undefined;
+      }
```

Comment:
The recommended way of checking types based on TS docs, still feels kind of hacky though.
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates